### DISCUSSION TITLE
How do I add the Zigpy code I have changed to Home Assistant and control Zha device?

### DISCUSSION THREAD

**Discussion by hwzolin** [zigpy/zigpy#1094] [Q&A] [answered]
Hello everyone, I have encountered a problem now.
Because it is the first time, I don't know how to use it. I added code related to our company OTA in ZIGPY.
I want to put the modified Zigpy code in the Home Assistant to run to see if it can control the device added by ZHA normally for OTA. How should I operate？

**Comment by Hedda** [comment, marked as answer]
FYI, there is some related discussion here about editing/chaning in/to dependecy libraries in Docker containers here -> https://github.com/zigpy/zha-device-handlers/discussions/693

**Reply by hwzolin** [reply]
Thank you

**Comment by Hedda** [comment]
Also check out OTAManager proof of concept code in https://github.com/zigpy/zigpy/pull/1102
